src/components/common/layout/Layout.jsx

```diff
@@ -5,6 +5,7 @@ import { IntlProvider } from 'react-intl';
 import SiteHeader from '@edx/frontend-component-site-header';
 import SiteFooter from '@edx/frontend-component-footer';
 import { connect } from 'react-redux';
+import { Link } from "@reach/router";
```

Comment:
Out of curiosity, did you try or consider using Gatsby's router that wraps @reach/router? The benefit of using Gatsby's wrapper around @reach/router is that you get prefetching when switching between routes (https://www.gatsbyjs.org/docs/gatsby-link/).